### DISCUSSION TITLE
brew upgrade swi-prolog fails from source

### DISCUSSION THREAD

**Discussion by Yann-R** [Homebrew/discussions#4163] [Everyday usage] [answered]
### Output of `brew config`

```shell
$ brew config
HOMEBREW_VERSION: 3.6.20-169-g856a347
ORIGIN: https://github.com/Homebrew/brew
HEAD: 856a347bfd0aa05ad644a864edf311ea1f7cbbf8
Last commit: 2 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 9d2fca0f48c1a7966a141aad47456dfb92bc934c
Core tap last commit: 8 days ago
Core tap branch: master
HOMEBREW_PREFIX: /opt/brew
HOMEBREW_REPOSITORY: /opt/brew/Homebrew
HOMEBREW_CELLAR: /opt/brew/Cellar
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.1QywUNaafY/org.xquartz:0
HOMEBREW_INSTALL_FROM_API: set
HOMEBREW_MAKE_JOBS: 16
Homebrew Ruby: 2.6.10 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 16-core 64-bit kabylake
Clang: 14.0.0 build 1400
Git: 2.39.1 => /opt/brew/bin/git
Curl: 7.79.1 => /usr/bin/curl
macOS: 12.6.2-x86_64
CLT: 14.2.0.0.1.1668646533
Xcode: 14.2
```

### Output of `brew doctor`

```shell
$ brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed kegs have no formulae!
This means they were either deleted or installed manually.
You should find replacements for the following formulae:
  xtail
  growly
  colorsvn
  contacts
  godep
  csv-fix
  szip

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  brew-pip
  gcc@6
  godep
  isl@0.18
  q
  rdesktop

Warning: Unbrewed header files were found in /usr/local/include.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected header files:
  /usr/local/include/ublio.h

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  unison

Warning: Some installed formulae are not readable:
  growly: wrong number of arguments (given 1, expected 0)

Warning: Your Homebrew's prefix is not /usr/local.

Many of Homebrew's bottles (binary packages) can only be used with the default prefix.
Consider uninstalling Homebrew and reinstalling into the default prefix.
It is expected behaviour that some formulae will fail to build in this unsupported configuration.
It is expected behaviour that Homebrew will be buggy and slow.
Do not create any issues about this on Homebrew's GitHub repositories.
Do not create any issues even if you think this message is unrelated.
Any opened issues will be immediately closed without response.
Do not ask for help from MacHomebrew on Twitter.
You may ask for help in Homebrew's discussions but are unlikely to receive a response.
Try to figure out the problem yourself and submit a fix as a pull request.
We will review it but may or may not accept it.
```

### Description of issue

$ brew upgrade swi-prolog
==> Upgrading 1 outdated package:
swi-prolog 9.0.3 -> 9.0.4
==> Fetching swi-prolog
Warning: Building swi-prolog from source as the bottle needs:
 HOMEBREW_CELLAR: /usr/local/Cellar (yours is /opt/brew/Cellar)
 HOMEBREW_PREFIX: /usr/local (yours is /opt/brew)
==> Downloading https://www.swi-prolog.org/download/stable/src/swipl-9.0.4.tar.gz
######################################################################## 100.0%
==> Upgrading swi-prolog
  9.0.3 -> 9.0.4 

==> cmake -S . -B build -DSWIPL_PACKAGES_JAVA=OFF -DSWIPL_PACKAGES_X=OFF -DCMAKE_INSTALL_RPATH=@loader_path -DCMAKE_INSTALL_PREFIX=/opt/brew/Cellar/swi-prolog/9.0.4/libexec
Last 15 lines from /Users/ricquebo/Library/Logs/Homebrew/swi-prolog/01.cmake:

  	 looks like Apple readline.  We need GNU readline.  Dropping

  	 the readline interface

-- Configuring done
CMake Error: AUTOUIC for target swipl-win: Test run of "uic" executable "/opt/brew/share/qt/libexec/uic" failed.
/opt/brew/share/qt/libexec/uic -h

dyld[86283]: Library not loaded: '/opt/brew/opt/icu4c/lib/libicui18n.71.dylib'
  Referenced from: '/opt/brew/Cellar/qt/6.4.1_1/lib/QtCore.framework/Versions/A/QtCore'
  Reason: tried: '/opt/brew/opt/icu4c/lib/libicui18n.71.dylib' (no such file), '/usr/local/lib/libicui18n.71.dylib' (no such file), '/usr/lib/libicui18n.71.dylib' (no such file), '/opt/brew/Cellar/icu4c/72.1/lib/libicui18n.71.dylib' (no such file), '/usr/local/lib/libicui18n.71.dylib' (no such file), '/usr/lib/libicui18n.71.dylib' (no such file)
Subprocess aborted
CMake Generate step failed.  Build files cannot be regenerated correctly.

Do not report this issue to Homebrew/brew or Homebrew/core!

**Comment by SMillerDev** [comment, marked as answer]
```
dyld[86283]: Library not loaded: '/opt/brew/opt/icu4c/lib/libicui18n.71.dylib'
Referenced from: '/opt/brew/Cellar/qt/6.4.1_1/lib/QtCore.framework/Versions/A/QtCore'
```
Your QT install references an old icu4c install that I guess has been upgraded since. Update/rebuild QT and then try again.

**Comment by Yann-R** [comment]
You must be right.
Forcing the upgrade of formula `qt` solved the failure concerning `swi-prolog` 👍
(Likely to be related to my other (solved) issue about upgrading `libx11`, but strange that `swi-prolog` formula started installing with outdated dependencies).
Thanks a lot.
